test/jdk/javax/swing/JEditorPane/bug4694598.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2002, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 4694598
+ * @key headful
+ * @summary JEditor pane throws NullPointerException on mouse movement.
+ * @run main bug4694598
+ */
+
+import java.awt.AWTException;
+import java.awt.Robot;
+import java.io.IOException;
+import java.io.Writer;
+import java.lang.reflect.InvocationTargetException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import javax.swing.JEditorPane;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+
+public class bug4694598 {
+    JFrame frame;
+    volatile int bottom;
+    final Path frameContentFile = Path.of("FrameContent.html");
+
+    public void setupGUI() {
+        frame = new JFrame("Test 4694598");
+        JEditorPane jep = new JEditorPane();
+        jep.setEditable(false);
+        frame.getContentPane().add(jep);
+        frame.setLocation(50, 50);
+        frame.setSize(400, 400);
+
+        String frameContentString = "<HTML><BODY>\n" +
+                "    Frame content.\n" +
+                "</BODY></HTML>";
+        try (Writer writer = Files.newBufferedWriter(frameContentFile)) {
+            writer.write(frameContentString);
+        } catch (IOException ioe){
+            throw new RuntimeException("Could not create html file to embed", ioe);
+        }
+        URL frameContentUrl;
+        try {
+            frameContentUrl = frameContentFile.toUri().toURL();
+        } catch (MalformedURLException mue) {
+            throw new RuntimeException("Can not convert path to URL", mue);
+        }
+        jep.setContentType("text/html");
+        String html = "<HTML> <BODY>" +
+                "<FRAMESET cols=\"100%\">" +
+                "<FRAME src=\"" + frameContentUrl + "\">" +
```

Comment:
```suggestion
                "<FRAME src=\"" + frameContentFile.toUri()+ "\">" +
```
Isn't it enough? Alternatively, `"file:/" + frameContentFile.toAbsolutePath()` produces the same result as `frameContentFile.toUri().toURL()`.

Another option is to convert the `Path` to `URL` in the `main` method before calling `setupGUI` and remove try-catch blocks.

Reply:
Okay.

Without the try-catch blocks, the code is simpler… Ensuring the updated test still reproduces the original issue is always a good thing to do, but running the updated with an old version of Java often requires modifying the source code again.